src/GoodsList.tsx

```diff
@@ -1,16 +1,20 @@
-import React from 'react';
+import React, { memo } from 'react';
 import { Good } from './types/Good';
 
 type Props = {
   goods: Good[]
 };
 
-export const GoodsList: React.FC<Props> = ({ goods }) => (
+export const GoodsList: React.FC<Props> = memo(({ goods }) => (
```

Comment:
React memo redundant here, because goods changed every time when you click on the button(getAll, get5...)